### PR TITLE
[UXIT-1261] Include Careers link in Nav and Footer

### PR DESCRIPTION
**Pull request by mirhamasala**
## 📝 Description

This pull request introduces a new "Careers" link to the navigation items. It also involves a refactor of the navigation components in the application, splitting the existing monolithic components into smaller, more manageable ones.

### Component Refactoring:

* **Mobile Navigation:**
  - Created `IconButton` component (`src/app/_components/Navigation/MobileNavigation/IconButton.tsx`).
  - Created `LinkItem` component (`src/app/_components/Navigation/MobileNavigation/LinkItem.tsx`).
  - Created `NestedMenu` component (`src/app/_components/Navigation/MobileNavigation/NestedMenu.tsx`).
  - Refactored `MobileNavigation` to use the new components (`src/app/_components/Navigation/MobileNavigation/MobileNavigation.tsx`).

* **Desktop Navigation:**
  - Created `MainNavItem` component (`src/app/_components/Navigation/DesktopNavigation/MainNavItem.tsx`).
  - Created `SubNavItem` component (`src/app/_components/Navigation/DesktopNavigation/SubNavItem.tsx`).
  - Refactored `DesktopNavigation` to use the new components and updated styles (`src/app/_components/Navigation/DesktopNavigation/DesktopNavigation.tsx`). [[1]](diffhunk://#diff-2cb202ce93235be987f7274e9575784c2f9dc2f380bbf1a1eda1de5ae9f0ff88L3-R24) [[2]](diffhunk://#diff-2cb202ce93235be987f7274e9575784c2f9dc2f380bbf1a1eda1de5ae9f0ff88L126-R46) [[3]](diffhunk://#diff-2cb202ce93235be987f7274e9575784c2f9dc2f380bbf1a1eda1de5ae9f0ff88R65) [[4]](diffhunk://#diff-2cb202ce93235be987f7274e9575784c2f9dc2f380bbf1a1eda1de5ae9f0ff88R106-R120)

### Navigation Items Update:

* **New "Careers" Link:**
  - Added "Careers" link to the `getInvolvedItems` array in `navigationItems` (`src/app/_constants/navigationItems.ts`).
  - Updated `siteMetadata` to include the "Careers" URL (`src/app/_constants/siteMetadata.ts`).
  - Updated `navigationData` to include the "Careers" item (`src/app/_data/navigationData.ts`).

## 📸 Screenshots
CleanShot 2024-10-22 at 09 53 56@2x

CleanShot 2024-10-21 at 16 06 58@2x

CleanShot 2024-10-21 at 15 57 42@2x

### CODE REVIEW

File: src/app/_components/Navigation/MobileNavigation/LinkItem.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        className={clsx(
+          linkBaseStyles,
+          "relative before:absolute before:inset-0 before:-m-3.5 before:content-['']",
+        )}
+        onClick={() => setOpen(false)}
+      >
+        {label}
+      </Link>
+      {isExternal && (
+        <Icon size={16} component={ArrowUpRight} color="brand-400" />
```
Comment (barbaraperic) [inline src/app/_components/Navigation/MobileNavigation/LinkItem.tsx:45]: We could extract this as it's own component? I think we use it in a couple of places. So we can have something like this:
`{isExternal && <ExternalLinkIcon />}`

**Review by barbaraperic — APPROVED** [review]

**Review by CharlyMartin — COMMENTED** [review]
Nice refactor Mirha! The code reads nicer 👌

I added some non-blocking comments and questions. Other than than, it works great!

File: src/app/_components/Navigation/DesktopNavigation/MainNavItem.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  isActive = false,
+}: MainNavItemProps) {
+  return (
+    <li>
+      <Link href={href} className={getMainNavItemStyles(isActive)}>
+        {label}
+      </Link>
+    </li>
+  )
+}
```
[the remark this answers is not in the document]
Comment (CharlyMartin) [reply src/app/_components/Navigation/DesktopNavigation/MainNavItem.tsx:29]: Scratch that, here it matters to make `getMainNavItemStyles` happy

File: src/app/_components/Navigation/MobileNavigation/IconButton.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  return (
+    <button
+      aria-label={label}
+      className="grid size-12 place-items-center rounded-lg border border-brand-300 text-brand-300 focus:brand-outline"
+      onClick={onClick}
+    >
+      <Icon size={20} component={icon} />
+    </button>
+  )
+}
```
[the remark this answers is not in the document]
Comment (mirhamasala) [reply src/app/_components/Navigation/MobileNavigation/IconButton.tsx:19]: I like that.

File: src/app/_components/Navigation/MobileNavigation/IconButton.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+type IconButtonProps = {
+  icon: IconProps['component']
+  label: string
+  onClick: React.ComponentPropsWithoutRef<'button'>['onClick']
+}
+
+export function IconButton({ icon, label, onClick }: IconButtonProps) {
+  return (
+    <button
```
[the remark this answers is not in the document]
Comment (mirhamasala) [reply src/app/_components/Navigation/MobileNavigation/IconButton.tsx:11]: Okay!

File: src/app/_components/Navigation/MobileNavigation/LinkItem.tsx (hunk 1 above)
Comment (mirhamasala) [reply src/app/_components/Navigation/MobileNavigation/LinkItem.tsx:45]: We might do this at some point. Right now, it occurs in two places and is one line of code so feels a bit too early to refactor. 🙏🏼

File: src/app/_components/Navigation/DesktopNavigation/SubNavItem.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  return external ? (
+    <a href={href} {...commonProps} rel="noopener noreferrer">
+      <div className="inline-flex items-center gap-1">
+        <p className="font-bold">{label}</p>
+        <span className="text-brand-400 group-hover:text-brand-100">
+          <Icon component={ArrowUpRight} size={20} />
+        </span>
+      </div>
+      {description && <p className="mt-1 text-brand-300">{description}</p>}
+    </a>
```
[the remark this answers is not in the document]
Comment (mirhamasala) [reply src/app/_components/Navigation/DesktopNavigation/SubNavItem.tsx:53]: Sure!

File: src/app/_components/Navigation/DesktopNavigation/MainNavItem.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+export function getMainNavItemStyles(isActive: boolean, isPopover = false) {
```
[the remark this answers is not in the document]
Comment (mirhamasala) [reply src/app/_components/Navigation/DesktopNavigation/MainNavItem.tsx:31]: I agree.

File: src/app/_data/navigationData.ts [hunk 7]
```diff
@@ -1,32 +1,15 @@
 import { PATHS } from '@/constants/paths'
-import { FILECOIN_DOCS_URLS } from '@/constants/siteMetadata'
+import {
+  FILECOIN_DOCS_URLS,
+  FILECOIN_FOUNDATION_URLS,
+} from '@/constants/siteMetadata'
```
[the remark this answers is not in the document]
Comment (mirhamasala) [reply src/app/_data/navigationData.ts:6]: Yes, you are right. I have a WIP CR - Will review it tomorrow. Sidenote: All your review comments are valid and valuable; however, they are very much scout observations - We should probably discuss how we handle these in the future. 🙏🏼

[the remark this answers is not in the document]
Comment (CharlyMartin) [reply src/app/_data/navigationData.ts:6]: Yes, fair point! We can add to the [followup dev work](https://www.notion.so/filecoin/FF-Followup-Dev-Work-d6616163f43c4e7db78edf218cebcc8d?pvs=4) ticket? Though I don't feel like we check it often.

**Comment by CharlyMartin** [comment]
Super cool what you did here @mirhamasala, much cleaner 👏✨🎉

I made some tiny changes for slightly improved readability (in my opinion)

**Review by CharlyMartin — APPROVED** [review]

File: src/app/_data/navigationData.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (CharlyMartin) [reply src/app/_data/navigationData.ts:6]: Yes I meant "I" really 😅

**Comment by mirhamasala** [comment]
@CharlyMartin - Love it, team. Let's go! ❤️

>I made some tiny changes for slightly improved readability (in my opinion)

**Pull request merged**